ballerina/vector_store.bal

```diff
@@ -66,41 +65,48 @@ public isolated class VectorStore {
             weaviate:Object[] objects = [];
             foreach ai:VectorEntry entry in entries.cloneReadOnly() {
                 ai:Embedding embedding = entry.embedding;
+                weaviate:PropertySchema properties = entry.chunk.metadata !is () ? 
+                    check entry.chunk.metadata.cloneWithType() : {};
+                properties[self.chunkFieldName] = entry.chunk.content;
+                properties["type"] = entry.chunk.'type;
+
                 if embedding is ai:Vector {
                     objects.push({
                         'class: self.config.collectionName,
                         id: entry.id,
                         vector: embedding,
-                        properties: {
-                            "type": entry.chunk.'type,
-                            [self.chunkFieldName]: entry.chunk.content
-                        }
+                        properties
                     });
                 }
                 // TODO: Add support for sparse and hybrid embeddings
                 // Weaviate does not support custom sparse or hybrid embeddings directly
                 // Need to convert them to dense vectors before adding to Weaviate
             }
-            weaviate:ObjectsGetResponse[]|error result = self.weaviateClient->/batch/objects.post({
+            weaviate:ObjectsGetResponse[] _ = check self.weaviateClient->/batch/objects.post({
                 objects
             });
-            if result is error {
-                return error("Failed to add vector entries", result);
-            }
+        } on fail error err {
+            return error("Failed to query vector store", err);
         }
     }
 
     # Deletes a vector entry from the Weaviate vector store.
     #
-    # + id - The ID of the vector entry to delete
+    # + ids - ID/s of the vector entries to delete
     # + return - An `ai:Error` if the deletion fails, otherwise returns `()`
-    public isolated function delete(string id) returns ai:Error? {
+    public isolated function delete(string|string[] ids) returns ai:Error? {
         lock {
             string path = self.config.collectionName;
-            http:Response|error result = self.weaviateClient->/objects/[path]/[id].delete();
-            if result is error {
-                return error("Failed to query vector store", result);
+            if ids is string[] {
+                foreach string id in ids.cloneReadOnly() {
+                    ai:Error? result = deleteById(id, path, self.weaviateClient);
```

Comment:
doesn't weaviate has a batch delete API?

Reply:
The weaviate REST API collection does not include any API for batch delete. Used a transaction block to handle the scenario